platform/ui-next/src/components/DataRow/DataRow.tsx

```diff
@@ -7,193 +7,274 @@ import {
   DropdownMenuItem,
 } from '../../components/DropdownMenu';
 import { Icons } from '../../components/Icons/Icons';
-import {
-  Tooltip,
-  TooltipTrigger,
-  TooltipContent,
-  TooltipProvider,
-} from '../../components/Tooltip/Tooltip';
+import { Tooltip, TooltipTrigger, TooltipContent } from '../../components/Tooltip/Tooltip';
+import sanitizeHtml from 'sanitize-html';
 
 interface DataRowProps {
   number: number;
-  title: string;
+  disableEditing: boolean;
   description: string;
-  optionalField?: string;
-  colorHex?: string;
-  details?: string;
-  series?: string;
-  actionOptions: string[];
-  onAction: (action: string) => void;
+  details?: { primary: string[]; secondary: string[] };
+  //
   isSelected?: boolean;
   onSelect?: () => void;
+  //
+  isVisible: boolean;
+  onToggleVisibility: () => void;
+  //
+  isLocked: boolean;
+  onToggleLocked: () => void;
+  //
+  title: string;
+  onRename: () => void;
+  //
+  onDelete: () => void;
+  //
+  colorHex?: string;
+  onColor: () => void;
 }
 
 const DataRow: React.FC<DataRowProps> = ({
```

Comment:
Can you add documentation as to what this is?  DataRow is pretty general, but this class seems pretty specific as to what it displays.  I certainly would have no idea as to how to use DataRow without reading the code and figuring that out is fairly hard.